### ISSUE TITLE
[RARELY] problem with flyspeed 0

### ISSUE THREAD

**Issue by smscentras** [Zrips/CMI#8849]
### Description of Issue

Sometimes players can't fly, because flyspeed is 0
Don't know why and when, but problem exists.

### Version Information

```shell
[13:32:05 INFO]: -------------------------------------------
[13:32:05 INFO]: CMI: 9.7.4.0 SqLite
[13:32:05 INFO]: CMILib: 1.5.0.1
[13:32:05 INFO]: Server: Purpur(2176) 1.20.4-R0.1-SNAPSHOT-
[13:32:05 INFO]: CMI economy: Enabled Vault: 1.7.3-CMI CMI Chat: Disabled
[13:32:05 INFO]: Modules -> 49 enabled 17 disabled: spawnerProximity, tablist, moneyCheque, skin, jail, spawners, flightCharge, deathMessages, disabledEnchants, worldLimitsElytra, worldLimits, chatBubble, playerCombat, votifier, shulkerBackpack, spawnerCharge, namePlates
[13:32:05 INFO]: -------------------------------------------
```

### Errors

```shell
No errors
```

### Relevant Plugins

Vault 1.7.3-CMI

### Agreements

- [X] My server is supported by CMI.
- [X] My version of CMI at the time of this report is up to date.
- [X] I have searched the github and asked around before making this report.

**Comment by Zrips** [comment]
Do you use CMI for flight/walk speed adjustments?

**Comment by smscentras** [comment]
> Do you use CMI for flight/walk speed adjustments?

Yes, in my server players flying with /CMI fly command. But they are'nt have /cmi flyspeed command.

**Comment by Zrips** [comment]
Some login/password plugin tend to use option to set players walk and fly speed to 0 to prevent their movement until their log into game. CMI itself doesn't have a code which would set players fly or walk speed to 0 or to any other value if its not been called specifically via command. So you might want to look into different plugins which manage players fly speeds for potential issue. Closing ticket for now as it seems to be caused by different plugin.

**Comment by smscentras** [comment]
> Some login/password plugin tend to use option to set players walk and fly speed to 0 to prevent their movement until their log into game. CMI itself doesn't have a code which would set players fly or walk speed to 0 or to any other value if its not been called specifically via command. So you might want to look into different plugins which manage players fly speeds for potential issue. Closing ticket for now as it seems to be caused by different plugin.

BUT why not allways, then?

**Comment by Zrips** [comment]
Because its just a bug which happens in some instances. There is a single instance in entire CMI where flight speed is being adjusted and its inside flyspeed command which will never be triggered without triggering that command. So issue should be somewhere else. 
Like I mentioned flyspeed usually gets set to 0 by login plugins to prevent player movement until he is fully verified. Similar thing could be done by some minigame plugins which blocks player movement for short period of times or something. You might want to just keep an eye out for anything what could cause this issue and ask around what players have been doing before they got their speed set to 0 just to get better idea
